### PR TITLE
use zip

### PR DESCRIPTION
**Pull request by kevinsung**

### CODE REVIEW

File: qiskit_addon_dice_solver/dice_solver.py [hunk 1]
```diff
@@ -595,13 +595,12 @@ def _construct_ci_vec_from_addresses_amplitudes(
 
     addr_map = {uni_addr: i for i, uni_addr in enumerate(uniques)}
 
-    for idx, address in enumerate(addresses):
```
Comment (caleb-johnson) [inline qiskit_addon_dice_solver/dice_solver.py:598]: I'm surprised the linter didn't complain about the unused `idx` variable here

Comment (caleb-johnson) [reply qiskit_addon_dice_solver/dice_solver.py:598]: Oops, I see it below now :)

Comment (kevinsung) [reply qiskit_addon_dice_solver/dice_solver.py:598]: It was used, in line 604.

**Review by caleb-johnson — APPROVED** [review]
LGTM! Thanks

**Pull request merged**